conf/standalone.conf

```diff
@@ -29,6 +29,11 @@ metadataStoreUrl=
 # Configuration file path for metadata store. It's supported by RocksdbMetadataStore and EtcdMetadataStore for now
 metadataStoreConfigPath=
 
+# Whether we should enable fsync for local metadata store. It's supported by RocksdbMetadataStore for now
+# If this flag is true, metadata writes will be slower.
+# If this flag is false, and the machine crashes, some recent metadata writes may be lost.
+# Note that if it is just the process that crashes (i.e., the machine does not reboot), no writes will be lost even if it is false.
+metadataFsyncEnabled=true
```

Comment:
Maybe we don't need to expose it to users?
We just want to improve the test.
Suppose we allow users to configure with false. The machine crash might lead to the metadata being corrupted.

Reply:
Or we can expose it until we need it. I haven't seen any issues or discussions about this part.
And batch operations might be another choice to improve the rocksdb metadata performance.

Reply:
> Maybe we don't need to expose it to users? We just want to improve the test. Suppose we allow users to configure with false. The machine crash might lead to the metadata being corrupted.

Rocksdb still can sync `wal`/`memtable` file by `wal_bytes_per_sync` and `bytes_per_sync` config, I think when setting `wal_bytes_per_sync=1` metadata no write will be lost even if it is false. 

ref: https://github.com/facebook/rocksdb/wiki/WAL-Performance

Reply:
> Or we can expose it until we need it.

+1, we should be more cautious about exposing more configs to users. I think we can keep the config of `MetadataStoreConfig` and remove the one in `ServiceConfiguration`. It would be enough to solve the unit test case.

Reply:
> > Whether we should automatically set wal_bytes_per_sync=1 and set WriteOptions.sync=false ?
> 
> IIUC, `WriteOptions.sync` only applies `fsync` on WAL. So there is no need to config `wal_bytes_per_sync`.

In my local test, if enable `WriteOptions.sync`, `PulsarLedgerIdGeneratorTest.testGenerateLedgerId` will timeout but only set `wal_bytes_per_sync=1 ` it does not timeout

Reply:
> > Or we can expose it until we need it.
> 
> +1, we should be more cautious about exposing more configs to users. I think we can keep the config of `MetadataStoreConfig` and remove the one in `ServiceConfiguration`. It would be enough to solve the unit test case.

OK.